### ISSUE TITLE
[Bug]: react/prop-types fails when using custom FunctionComponent type

### ISSUE THREAD

**Issue by Kyzegs** [jsx-eslint/eslint-plugin-react#3625]
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Description Overview

As the title states, thr `react/prop-types` rule fails when using a custom `FunctionComponent` type.

```ts
export type FC<P = {}> = React.FC<P> & {
  layout?: (page: React.ReactNode) => React.JSX.Element
};
```

![image](https://github.com/jsx-eslint/eslint-plugin-react/assets/45851377/243b1b8b-cb2c-4f37-96b1-e0470823b84d)

### Expected Behavior

I'd expect that this should work as it extends the `React.FC` type.

### eslint-plugin-react version

v7.32.2

### eslint version

v8.43.0

### node version

v16.17.0
